### PR TITLE
Creat the sign up page

### LINKED ISSUE #16: Sign up page
**Task title**
Sign up page

**Task description**
the Sign-Up page will support multiple sign-up methods including email, Facebook, and Google authentication using Firebase. Users will have the flexibility to choose their preferred method, making the sign-up process more accessible and user-centric.  

**Subtasks**
- [x] Email Sign-up : Implement Firebase's email/password authentication. Ensure that the user receives a verification email and that we securely store user data in Firebase.

- [x] Facebook Sign-up : Integrate Firebase with Facebook authentication. This will enable users to sign up using their Facebook accounts. This should be a seamless process and provide us with basic user data.

- [x] Google Sign-up : Implement Firebase's Google authentication, giving users the option to use their Google accounts for sign-up. Securely manage the obtained user data.

- [ ] User Data Consistency : Ensure that, regardless of the sign-up method, user data is stored consistently in Firebase.

- [x] Sign up thanks page

- [x] Add translation

- [x] Responsive design

**Screenshots**
sign up
signup thanks

**Link to the component on Figma**
https://www.figma.com/file/4OUnD74RgIW6tsUFdb7gcW/Untitled?type=design&node-id=2-6529&mode=design&t=3WJyDJ9tmMfJOBVl-4

### PR DESCRIPTION
**Pull request by Lilicod**
This pull request introduces the signup page. Here's an overview of the changes:

**1. Responsive Design:**

**2. Multiple Signup Methods:**
Users can now sign up through three different methods:

- Email and Password
- Google
- Facebook

**3. User Profile Page:**
After successful signup, users can access their profile page, which displays their name and email address.

Capture d’écran 2023-10-23 182510
Capture d’écran 2023-10-23 182416
Capture d’écran 2023-10-23 182328

### CODE REVIEW

**Review by dzc0d3r — APPROVED** [review]
Amazing  Work :100: @Lilicod 

Thank You :1st_place_medal:

**Review by M07am3dM — APPROVED** [review]
Great work @Lilicod, well done.

**Review by meryem08 — APPROVED** [review]
Great work @Lilicod 👏👏

File: .husky/pre-commit [hunk 1]
```diff
@@ -3,4 +3,4 @@
 
 npm run test
 npm run format
-npx lint-staged
\ No newline at end of file
+# npx lint-staged
```
Comment (dzc0d3r) [inline .husky/pre-commit:6]: @Lilicod 
Please don't forget to uncomment this line before merging 
```suggestion
 npx lint-staged
```

**Review by dzc0d3r — CHANGES_REQUESTED** [review]

**Comment by HabchiSidAli** [comment]
Nicely done @Lilicod!

**Pull request merged**